src/Verbs/PushCommand.cs

```diff
@@ -0,0 +1,179 @@
+                    var fileInfo = new FileInfo(options.Package);
+                    var content = new StreamContent(fileInfo.OpenRead());
+                    content.Headers.ContentDisposition = new ContentDispositionHeaderValue("form-data")
+                    {
+                        Name = "file",
+                        FileName = fileInfo.Name
+                    };
+                    form.Add(content);
+                    form.Add(new StringContent(ParsePublishFlag(options.Publish)), "isCurrent");
+                    form.Add(new StringContent(options.DotNetVersion), "dotNetVersion");
+                    form.Add(new StringContent("package"), "fileType");
+                    form.Add(GetVersionCompatability(options.WorksWith), "umbracoVersions");
+
+                    var httpResponse = await client.PostAsync("/Umbraco/Api/ProjectUpload/UpdatePackage", form);
+                    if (httpResponse.StatusCode == HttpStatusCode.Unauthorized)
+                    {
+                        packageHelper.WriteError(Resources.Push_ApiKeyInvalid);
+                        Environment.Exit(5); // ERROR_ACCESS_DENIED
+                    }
+                    else if (httpResponse.IsSuccessStatusCode)
+                    {
+                        Console.WriteLine(Resources.Push_Complete);
+                        var apiReponse = await httpResponse.Content.ReadAsStringAsync();
+                        // Console.WriteLine(apiReponse);
+                    }
+                }
+            }
+            catch (HttpRequestException ex)
+            {
+                // Could get network error or our.umb down
+                Console.WriteLine(Resources.Error, ex);
+                throw;
+            }
+        }
+
+
+
+        /// <summary>
+        ///  returns the version compatability string for uploading the package
+        /// </summary>
+        /// <param name="worksWithString"></param>
+        /// <returns></returns>
+        private static StringContent GetVersionCompatability(string worksWithString)
+        {
+            // TODO: Workout how we can get a latest version from our ? 
+            // TODO: Maybe accept wild cards (8.* -> 8.5.0,8.4.0,8.3.0)
+            // TODO: Work like nuget e.g '> 8.4.0' 
+            var versions = worksWithString
+                                .Split(",", StringSplitOptions.RemoveEmptyEntries)
+                                .Select(x => new UmbracoVersion() { Version = x });
+
+            return new StringContent(JsonConvert.SerializeObject(versions));
+        }
+
+        private static string ParsePublishFlag(string publish)
+        {
+            if (bool.TryParse(publish, out bool result))
+            {
+                return result.ToString();
+            }
+
+            return false.ToString();
+        }
+
+        private class UmbracoVersion
+        {
+            public string Version { get; set; }
+            public string Name { get; set; }
```

Comment:
Name seems to not be used.

Reply:
yeah it was a copy of this model from our source - might be best to comment it out, but keep something - if it ever needs to go back in.

Reply:
done